examples/api/lib/material/menu_bar/menu_bar.1.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Example for MenuBar.adaptive.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() {
+  runApp(const MaterialApp(
+    title: 'Adaptive Menu Sample',
+    home: Material(child: Home()),
+  ));
+}
+
+enum MenuSelection {
+  edit('Edit'),
+  cut('Cut'),
+  copy('Copy'),
+  paste('Paste'),
+  file('File'),
+  open('Open'),
+  quit('Quit'),
+  save('Save'),
+  saveAs('Save As...');
+
+  const MenuSelection(this.label);
+  final String label;
+}
+
+
+class Home extends StatefulWidget {
+  const Home({super.key});
+
+  @override
+  State<Home> createState() => _HomeState();
+}
+
+class _HomeState extends State<Home> {
+  bool isPlatformMenu = false;
+  TextDirection textDirection = TextDirection.ltr;
+  bool enabled = true;
+  bool checked = false;
+
+  void _onSelected(MenuSelection item) {
+    debugPrint('Activated ${item.name}');
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final bool isMacOS = defaultTargetPlatform == TargetPlatform.macOS;
```

Comment:
use switch/case over defaultTargetPlatform so the analyzer tells us to revisit this place when we add a new platform?